src/compile/repeater.ts

```diff
@@ -20,40 +30,70 @@ export function replaceRepeaterInEncoding(encoding: Encoding<Field>, repeater: R
 /**
  * Replace repeater values in a field def with the concrete field name.
  */
-function replaceRepeaterInFieldDef(fieldDef: FieldDef<Field>, repeater: RepeaterValue): FieldDef<string> | null {
+function replaceRepeaterInFieldDef(fieldDef: LegendFieldDef<Field>, repeater: RepeaterValue): LegendFieldDef<string> | null {
   const field = fieldDef.field;
   if (isRepeatRef(field)) {
     if (field.repeat in repeater) {
-      return {
+      fieldDef = {
         ...fieldDef,
         field: repeater[field.repeat]
       };
     } else {
       log.warn(log.message.noSuchRepeatedValue(field.repeat));
+      // return null, meaning that the field def should be ignored
       return null;
     }
-  } else {
-    // field is not a repeat ref so we can just return the field def
-    return fieldDef as FieldDef<string>;
   }
+
+  const sort = fieldDef.sort;
+  if (sort && isSortField(sort) && isRepeatRef(sort.field)) {
+    if (sort.field.repeat in repeater) {
```

Comment:
This part is redundant with above, we should extract a more general `replaceRepeat` method that takes any object with a property that can be a `RepeatRef`. 

Something like
```
function replaceRepeat<T>(o: T, prop: keyof T, repeater: Repeater) {...}
```
may work (I haven't tried myself).

Reply:
Done